common/util/src/main/java/org/thingsboard/common/util/JacksonUtil.java

```diff
@@ -126,6 +126,14 @@ public static <T> T fromBytes(byte[] bytes, Class<T> clazz) {
         }
     }
 
+    public static <T> T fromBytes(byte[] bytes, TypeReference<T> valueTypeRef) {
+        try {
+            return bytes != null ? OBJECT_MAPPER.readValue(bytes, valueTypeRef) : null;
+        } catch (IOException e) {
+            throw new IllegalArgumentException("The given string value cannot be transformed to Json object: " + Arrays.toString(bytes), e);
```

Comment:
Please log only the first 1k (10k) chars or some limited length up to you. Some payloads like images or jsons may be few MegaBytes long and it floods or break the logs infrastructure.